modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ScalaAkkaClientCodegen.java

```diff
@@ -113,7 +113,7 @@ public ScalaAkkaClientCodegen() {
         typeMapping.put("binary", "File");
         typeMapping.put("number", "Double");
 
-        instantiationTypes.put("array", "ListBuffer");
+        instantiationTypes.put("array", "List");
```

Comment:
This is a breaking change, and won't work because templates expect mutation.

Reply:
When do we access generated array in templates? Do you mean tests? Or am I missing something?

Reply:
@fokot "templates" here doesn't just refer to the built-in templates under resources. openapi-generator supports user customized templates, so changing the above will break expectations of anyone with customized templates who expect mutability.  Sorry to not be more clear about that.

Also, `instantiationTypes` really only makes sense as something which can instantiate an object. You can see the difference in your test modifications, for example:

```diff
-        Assert.assertEquals(property1.defaultValue, "new ListBuffer[String]() ");
+        Assert.assertEquals(property1.defaultValue, "List.empty ");
```

This might be used in a class. Now, rather than an immutable property reference with a mutable data structure:

```
  prop: ListBuffer[String] = new ListBuffer[String]()
```

We have an immutable type:

```
   prop: List[String] = List.empty
```

while I agree that the latter is more idiomatic, and you can just use the copy method as case classes intend, anyone who has previously generated against openapi-generator will have to change their client call sites or communicate the change to client SDK users.

If possible, could we instead rollback this change in this PR and open another PR against the 5.0 branch for this change? There is a workaround in 4.x to specify the instantiation types and type mappings manually for anyone who wants the functionality you've implemented here. Unfortunately, doing the right think now means introducing a breaking change for those using the template.